client/genclients.go

```diff
@@ -94,17 +92,27 @@ func methodCode(op *spec.Operation, basePath, method, methodPath string) string
 	capOpID := swagger.Capitalize(op.ID)
 
 	buf.WriteString(swagger.InterfaceComment(method, methodPath, op) + "\n")
-	buf.WriteString(fmt.Sprintf("func (c Client) %s {\n", swagger.Interface(op)))
+	buf.WriteString(fmt.Sprintf("func (c *Client) %s {\n", swagger.Interface(op)))
 	buf.WriteString(fmt.Sprintf("\tpath := c.basePath + \"%s\"\n", basePath+methodPath))
 	buf.WriteString(fmt.Sprintf("\turlVals := url.Values{}\n"))
 	buf.WriteString(fmt.Sprintf("\tvar body []byte\n\n"))
 
 	buf.WriteString(fmt.Sprintf(buildRequestCode(op, method)))
 
 	buf.WriteString(fmt.Sprintf(`
+
 	// Add the opname for doers like tracing
 	ctx = context.WithValue(ctx, opNameCtx{}, "%s")
-	resp, err := c.requestDoer.Do(client, req.WithContext(ctx))
+	req = req.WithContext(ctx)
+	// Don't add the timeout in a "doer" because we don't want to call "defer.cancel()"
+	// until we've finished all the processing of the request object. Otherwise we'll cancel
+	// our own request before we've finished it.
+	if c.defaultTimeout != 0 {
+		ctx, cancel := context.WithTimeout(req.Context(), c.defaultTimeout)
+		defer cancel()
```

Comment:
I'm not totally sure what the Go http library does when this cancel gets called, but since we're done with the request I think it should be fairly safe. Let me know if you have any concerns. Otherwise I guess we just live with using extra memory...